openedx/core/djangoapps/discussions/models.py

```diff
@@ -240,6 +242,30 @@ class DiscussionsConfiguration(TimeStampedModel):
     Associates a learning context with discussion provider and configuration
     """
 
+    def user_permissions_for_course(self, course, user):
+        """
+        Return the user's permissions over the discussion configuration of the course.
+        """
+        return {
+            "change_provider": (
+                not course.start <= datetime.now(timezone.utc)
+                or GlobalStaff().has_user(user),
+                "Discussion provider cannot be changed after the course has started."
+                " Please reach out to partner support.",
```

Comment:
I think this is more developer-facing, so it should probably mention that the current user doesn't have permissions to perform this operation. 
```suggestion
                "Must be global staff to change discussion provider after the course has started."
```